### ISSUE TITLE
Improve the mapping/prefilling the new case/new sample forms when processing a DEMIS LabMessage [2]

### ISSUE THREAD

**Issue by markusmann-vg** [hzi-braunschweig/SORMAS-Project#4106]
### Situation Description
Currently some of the fields that are needed to create a new case or new sample from a DEMIS Lab Message have to be entered manually. 
The expectation is that we read from the demis lab message and prefill the fields.
If no mapping is possible, we have to add it to a list, which can be discussed with HZI and Gematik.

Like:
DEMIS => Swab // SORMAS: Throat swab & Nasal swab
Lab => described // SORMAS: other lab = take over the information

**Comment by mmarzaganov** [comment]
@markusmann-vg @kwa20 Please check and advise on the following:

Here is a list of fields that would definitely need to be mapped for autofill:

CT-Werte / CQ-Value (should be available in combination with PCR/qPCR)
Probeetnahmedatum / Date of Sample Collection
Probentyp / Type of Sample (within Lab Message referred to _Typ_ under _Probe_)
Art des Tests / Type of Test
Testergebnis / Result
Getestete Krankheit / Tested Disease
Probezustand / Specimen Condition

_Date Sample Received at Lab_ appears to be already mapped and prefilled. 

A couple of points that I am not sure about:

1. Ergebnis Verifiziert Von Laborabteilung / Result Verified by Lab Supervisor 
Does this de-factor apply when within the Lab Message the _Status_ (under _Erregernachweis_) is set to _final_?

2. Feld Probe-ID/Field Sample ID
Is that the same as _Meldungsnummer_ in the Lab Message? Or is it something else (e.g. for other countries)?

**Comment by vlad-ciucescu** [comment]
@markusmann-vg @kwa20 @Candice-Louw @ErikMarzHZI 
The following fields can be mapped easily, besides what is already available:

- Meldedatum
- person phone
- person email
- tested disease -> if it is COVID-19 only then it is simple, otherwise there doesn't seem to be any easy way to find out the disease from the message
- test result

The following fields need to be clarified:

- type of sample -> seems to be free text in some cases, in which case it cannot be mapped to the sormas field
![image](https://user-images.githubusercontent.com/65336966/108376837-10164100-720c-11eb-8d0a-55d5f2cf42da.png)
- lab -> 'Andere Einrichtung', but what should we put in the description. In SORMAS we have only one description field, in the messages we have a lot of data:
![image](https://user-images.githubusercontent.com/65336966/108377221-7a2ee600-720c-11eb-9b2a-ac584c15be5a.png)
- CT-Werte / CQ-Value -> please tell me which part of the message contains these values
- type of test -> I suppose it is the 'Methode' in the Erregernachweis. In this case it is the same problem as in the case of the type of sample, we cannot easily map those values to SORMAS values
![image](https://user-images.githubusercontent.com/65336966/108378176-718adf80-720d-11eb-9eeb-11b55365332e.png)
- Probeetnahmedatum -> each 'Hinweis' field comes as a free text, meaning that we would have to extract only the part containing the date from that text. This can be problematic, since the format ca vary arbitrarily e.g `Entnahmedatum: 03.02.2021`, `Entnahmedatum:03.02.2021`, `Entnahmedatum ist 03.02.2021`, `Entnahmedatum: 2021-02-03`. Unless we receive this value in a separate field, it is very error prone to obtain it from the Hinweis. For the same reason it is difficult to make this date bold.

Please also clarify the other three points you mentioned.

Best regards,
Vlad

**Comment by FredrikSchaeferVitagroup** [comment]
@Candice-Louw , @kwa20 : Regarding the test result: We were told several times by HZI that we shall not directly map the test result, as some test results marked positive in a demis notification have to be considered not positive sometimes. Has this changed?

**Comment by FredrikSchaeferVitagroup** [comment]
What is making the processing of labMessages cumbersome are especially the fields a are not prefilled and mandatory. Those are (in addition to the ones already mentioned):

- Responsible Region
- Responsible District

Can we assume them to default to the region and district of the receiving health department? I remember that some notifications are transmitted to the wrong health department via DEMIS.

**Comment by FredrikSchaeferVitagroup** [comment]
### scope 

- [ ] Bundle.meta.lastUpdated = PathogenTest.ReportDate
- [ ] Bundle.entry.ressource.observation.interpretation.code = PathogenTest.testResult [POS = POSITIVE, NEG = NEGATIVE, IND = INDETERMINATE]
- [ ] Prefill Sample.SpecimenCondition always with ADEQUATE
- [ ] Prefill PathogenTest.testResultVerified always with true
- [ ] Prefill PathogenTest.testedDisease always with CORONAVIRUS
- [ ] Person.emailAddress
- [ ] Person.Phone

### out of scope

-  Mapping of Regions and Disctricts (see https://github.com/hzi-braunschweig/SORMAS-Project/issues/4508)

### Not (yet) to be mapped

- PathogenTest.testType (no specification yet)
- Sample.sampleMaterial (no specification yet)
- PathogenTest.CQVaule (inclear how to extract the value from string)
- Case/Contact.reportDate (unclear which date value to relate to it)
- Sample.sampleDateTime (unclear which date value to relate to it)

**Comment by Candice-Louw** [comment]
@JaquM - further input/feedback most welcome, please. :)

**Comment by vlad-ciucescu** [comment]
Should be reviewed after #4017

**Comment by vlad-ciucescu** [comment]
To validate:
Check the fields noted by Fredrik
![image](https://user-images.githubusercontent.com/65336966/112334490-23fa0a80-8cc4-11eb-8a23-98df3b44c7d3.png)

Note that some lab messages don't have all these fields. Also, many fields are freetext and may not be taken even if they look valid.
Sample material is prefilled separately in #4773
